packages/runner/lib/sdk/locks.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable @typescript-eslint/require-await */
+import type { Result } from '@nangohq/utils';
+import { Err, Ok } from '@nangohq/utils';
+
+interface Lock {
+    key: string;
+    owner: string;
+    expiresAt: Date;
+}
+
+export class Locks {
+    private store = new Map<string, Lock>();
+    private mutex = new Map<string, boolean>();
+
+    public async tryAcquireLock({ owner, key, ttlMs }: { owner: string; key: string; ttlMs: number }): Promise<Result<boolean>> {
+        if (!owner || owner.length === 0 || owner.length > 255) {
+            return Err('Invalid lock owner (must be between 1 and 255 characters)');
+        }
+        if (!key || key.length === 0 || key.length > 255) {
+            return Err('Invalid lock key (must be between 1 and 255 characters)');
+        }
+        if (ttlMs <= 0) {
```

Comment:
I'm wondering if it's a foot gun to require a TTL? 
Customers will need to add something but they rarely know how long their syncs are running. If we consider that 99.9% of locks will go through `releaseAllLock` then I would not require a TTL or at least set a default to 24H

Reply:
I agree with you. I also think it is good that the user think twice about this feature and since it is a feature I thought it won't be a bad idea to require the ttl. Also I am not sure what would be a good default: 24h for syncs is fine but not for webhooks or actions. Let's see if I can set a default value based on the type of script

Reply:
even based on the type of script is hard to come up with a good default. It depends very much on what the script is doing. I am gonna stick with it and keep the ttl required. The ttl value is a tradeoff the customer has to make